include/deal.II/base/mpi.templates.h

```diff
@@ -460,77 +460,88 @@ namespace Utilities
     std::vector<T>
     compute_set_union(const std::vector<T> &vec, const MPI_Comm &comm)
     {
+      const auto allreduce = [](const auto &vec,
+                                const auto  comm,
+                                const auto &process) {
```

Comment:
These arguments are not self-descriptive. Can you give them actual, explicit types?